<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Universe of iSEE panels • iSEEu</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Universe of iSEE panels">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">iSEEu</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.17.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/iSEEu.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/iSEE/iSEEu/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Universe of iSEE panels</h1>
                        <h4 data-toc-skip class="author">Kevin
Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford,
Headington, Oxford OX3 7FY,
UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk" class="email">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                              <h4 data-toc-skip class="author">Federico
Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics
(IMBEI), MainzCenter for Thrombosis and Hemostasis (CTH),
Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a>
      </address>
                              <h4 data-toc-skip class="author">Charlotte
Soneson</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, BaselSIB
Swiss Institute of
Bioinformatics<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlottesoneson@gmail.com" class="email">charlottesoneson@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Aaron
Lun</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:infinite.monkeys.with.keyboards@gmail.com" class="email">infinite.monkeys.with.keyboards@gmail.com</a>
      
                  
            <h4 data-toc-skip class="date">10 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEu/blob/devel/vignettes/iSEEu.Rmd" class="external-link"><code>vignettes/iSEEu.Rmd</code></a></small>
      <div class="d-none name"><code>iSEEu.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> package
<span class="citation">(Rue-Albrecht et al. 2018)</span> provides a
general and flexible framework for interactively exploring
<code>SummarizedExperiment</code> objects. However, in many cases, more
specialized panels are required for effective visualization of specific
data types. The <em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em>
package implements a collection of such dedicated panel classes that
work directly in the <code>iSEE</code> application and can smoothly
interact with other panels. This allows users to quickly parametrize
bespoke apps for their data to address scientific questions of interest.
We first load in the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEu" class="external-link">iSEEu</a></span><span class="op">)</span></span></code></pre></div>
<p>All the panels described in this document can be deployed by simply
passing them into the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE()</a></code> function via the
<code>initial=</code> argument, as shown in the following examples.</p>
</div>
<div class="section level2">
<h2 id="differential-expression-plots">Differential expression plots<a class="anchor" aria-label="anchor" href="#differential-expression-plots"></a>
</h2>
<p>To demonstrate the use of these panels, we will perform a
differential expression analysis on the <em><a href="https://bioconductor.org/packages/3.20/airway" class="external-link">airway</a></em>
dataset with the <em><a href="https://bioconductor.org/packages/3.20/edgeR" class="external-link">edgeR</a></em>
package. We store the resulting statistics in the <code>rowData</code>
of the <code>SummarizedExperiment</code> so that it can be accessed by
<code>iSEE</code> panels.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>, samples<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="va">y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">y</span>, group<span class="op">=</span><span class="va">y</span><span class="op">$</span><span class="va">samples</span><span class="op">$</span><span class="va">dex</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">dex</span>, <span class="va">y</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html" class="external-link">estimateDisp</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFit.html" class="external-link">glmQLFit</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html" class="external-link">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags</a></span><span class="op">(</span><span class="va">res</span>, n<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span><span class="op">$</span><span class="va">table</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>, <span class="va">tab</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>The <code>MAPlot</code> class creates a MA plot, i.e., with the
log-fold change on the y-axis and the average expression on the x-axis.
Features with significant differences in each direction are highlighted
and counted on the legend. Users can vary the significance threshold and
apply <em>ad hoc</em> filters on the log-fold change. This is a subclass
of the <code>RowDataPlot</code> so points can be transmitted to other
panels as multiple row selections. Instances of this class are created
like:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ma.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAPlot-class.html">MAPlot</a></span><span class="op">(</span>PanelWidth<span class="op">=</span><span class="fl">6L</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">ma.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Fma.png"><!-- --></p>
<p>The <code>VolcanoPlot</code> class creates a volcano plot with the
log-fold change on the x-axis and the negative log-p-value on the
y-axis. Features with significant differences in each direction are
highlighted and counted on the legend. Users can vary the significance
threshold and apply <em>ad hoc</em> filters on the log-fold change. This
is a subclass of the <code>RowDataPlot</code> so points can be
transmitted to other panels as multiple row selections. Instances of
this class are created like:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vol.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>PanelWidth<span class="op">=</span><span class="fl">6L</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">vol.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Fvolcano.png"><!-- --></p>
<p>The <code>LogFCLogFCPlot</code> class creates a scatter plot of two
log-fold changes from different DE comparisons. This allows us to
compare DE results on the same dataset - or even from different
datasets, as long as the row names are shared. Users can vary the
significant threshold used to identify DE genes in either or both
comparisons. This is a subclass of the <code>RowDataPlot</code> so
points can be transmitted to other panels as multiple row selections.
Instances of this class are created like:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creating another comparison, this time by blocking on the cell line</span></span>
<span><span class="va">design.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">cell</span> <span class="op">+</span> <span class="va">dex</span>, <span class="va">y</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="va">y.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html" class="external-link">estimateDisp</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">design.alt</span><span class="op">)</span></span>
<span><span class="va">fit.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFit.html" class="external-link">glmQLFit</a></span><span class="op">(</span><span class="va">y.alt</span>, <span class="va">design.alt</span><span class="op">)</span></span>
<span><span class="va">res.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html" class="external-link">glmQLFTest</a></span><span class="op">(</span><span class="va">fit.alt</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tab.alt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags</a></span><span class="op">(</span><span class="va">res.alt</span>, n<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span><span class="op">$</span><span class="va">table</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>, alt<span class="op">=</span><span class="va">tab.alt</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span>,<span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lfc.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LogFCLogFCPlot-class.html">LogFCLogFCPlot</a></span><span class="op">(</span>PanelWidth<span class="op">=</span><span class="fl">6L</span>, YAxis<span class="op">=</span><span class="st">"alt.logFC"</span>, </span>
<span>    YPValueField<span class="op">=</span><span class="st">"alt.PValue"</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">lfc.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Flogfclogfc.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="dynamically-recalculated-panels">Dynamically recalculated panels<a class="anchor" aria-label="anchor" href="#dynamically-recalculated-panels"></a>
</h2>
<p>To demonstrate, we will perform a quick analysis of a small dataset
from the <em><a href="https://bioconductor.org/packages/3.20/scRNAseq" class="external-link">scRNAseq</a></em>
package. This involves computing normalized expression values and
low-dimensional results using the <em><a href="https://bioconductor.org/packages/3.20/scater" class="external-link">scater</a></em>
package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scRNAseq</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scRNAseq/man/ReprocessedData.html" class="external-link">ReprocessedAllenData</a></span><span class="op">(</span>assays<span class="op">=</span><span class="st">"tophat_counts"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce</span>, exprs_values<span class="op">=</span><span class="st">"tophat_counts"</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocSingular/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">sce</span>, ncomponents<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html" class="external-link">runTSNE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>The <code>DynamicReducedDimensionPlot</code> class creates a scatter
plot with a dimensionality reduction result, namely principal components
analysis (PCA),
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>-stochastic
neighbor embedding
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>t</mi><annotation encoding="application/x-tex">t</annotation></semantics></math>-SNE)
or uniform manifold and approximate projection (UMAP). It does so
dynamically on the subset of points that are selected in a transmitting
panel, allowing users to focus on finer structure when dealing with a
heterogeneous population. Calculations are performed using relevant
functions from the <em><a href="https://bioconductor.org/packages/3.20/scater" class="external-link">scater</a></em>
package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Receives a selection from a reduced dimension plot.</span></span>
<span><span class="va">dyn.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DynamicReducedDimensionPlot-class.html">DynamicReducedDimensionPlot</a></span><span class="op">(</span>Type<span class="op">=</span><span class="st">"UMAP"</span>, Assay<span class="op">=</span><span class="st">"logcounts"</span>,</span>
<span>    ColumnSelectionSource<span class="op">=</span><span class="st">"ReducedDimensionPlot1"</span>, PanelWidth<span class="op">=</span><span class="fl">6L</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># NOTE: users do not have to manually create this, just </span></span>
<span><span class="co"># copy it from the "Panel Settings" of an already open app.</span></span>
<span><span class="va">red.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/ReducedDimensionPlot-class.html" class="external-link">ReducedDimensionPlot</a></span><span class="op">(</span>PanelId<span class="op">=</span><span class="fl">1L</span>, PanelWidth<span class="op">=</span><span class="fl">6L</span>,</span>
<span>    BrushData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        xmin <span class="op">=</span> <span class="op">-</span><span class="fl">45.943</span>, xmax <span class="op">=</span> <span class="op">-</span><span class="fl">15.399</span>, ymin <span class="op">=</span> <span class="op">-</span><span class="fl">58.560</span>, </span>
<span>        ymax <span class="op">=</span> <span class="fl">49.701</span>, coords_css <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="fl">51.009</span>, </span>
<span>            xmax <span class="op">=</span> <span class="fl">165.009</span>, ymin <span class="op">=</span> <span class="fl">39.009</span>, </span>
<span>            ymax <span class="op">=</span> <span class="fl">422.009</span><span class="op">)</span>, coords_img <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="fl">66.313</span>, </span>
<span>            xmax <span class="op">=</span> <span class="fl">214.514</span>, ymin <span class="op">=</span> <span class="fl">50.712</span>, </span>
<span>            ymax <span class="op">=</span> <span class="fl">548.612</span><span class="op">)</span>, img_css_ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1.300</span>, </span>
<span>            y <span class="op">=</span> <span class="fl">1.299</span><span class="op">)</span>, mapping <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"X"</span>, y <span class="op">=</span> <span class="st">"Y"</span><span class="op">)</span>, </span>
<span>        domain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>left <span class="op">=</span> <span class="op">-</span><span class="fl">49.101</span>, right <span class="op">=</span> <span class="fl">57.228</span>, </span>
<span>            bottom <span class="op">=</span> <span class="op">-</span><span class="fl">70.389</span>, top <span class="op">=</span> <span class="fl">53.519</span><span class="op">)</span>, </span>
<span>        range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>left <span class="op">=</span> <span class="fl">50.986</span>, right <span class="op">=</span> <span class="fl">566.922</span>, </span>
<span>            bottom <span class="op">=</span> <span class="fl">603.013</span>, top <span class="op">=</span> <span class="fl">33.155</span><span class="op">)</span>, </span>
<span>        log <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>, direction <span class="op">=</span> <span class="st">"xy"</span>, </span>
<span>        brushId <span class="op">=</span> <span class="st">"ReducedDimensionPlot1_Brush"</span>, </span>
<span>        outputId <span class="op">=</span> <span class="st">"ReducedDimensionPlot1"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">red.panel</span>, <span class="va">dyn.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Fdynreddim.png"><!-- --></p>
<p>The <code>DynamicMarkerTable</code> class dynamically computes basic
differential statistics comparing assay values across groups of multiple
selections in a transmitting panel. If only the active selection exists
in the transmitting panel, a comparison is performed between the points
in that selection and all unselected points. If saved selections are
present, pairwise comparisons between the active selection and each
saved selection is performed and the results are combined into a single
table using the <code><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">findMarkers()</a></code> function from <em><a href="https://bioconductor.org/packages/3.20/scran" class="external-link">scran</a></em>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DynamicMarkerTable-class.html">DynamicMarkerTable</a></span><span class="op">(</span>PanelWidth<span class="op">=</span><span class="fl">8L</span>, Assay<span class="op">=</span><span class="st">"logcounts"</span>,</span>
<span>    ColumnSelectionSource<span class="op">=</span><span class="st">"ReducedDimensionPlot1"</span>,<span class="op">)</span></span>
<span></span>
<span><span class="co"># Recycling the reduced dimension panel above, adding a saved selection to</span></span>
<span><span class="co"># compare to the active selection.</span></span>
<span><span class="va">red.panel</span><span class="op">[[</span><span class="st">"SelectionHistory"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    BrushData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        xmin <span class="op">=</span> <span class="fl">15.143</span>, xmax <span class="op">=</span> <span class="fl">57.228</span>, ymin <span class="op">=</span> <span class="op">-</span><span class="fl">40.752</span>, </span>
<span>        ymax <span class="op">=</span> <span class="fl">25.674</span>, coords_css <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="fl">279.009</span>, </span>
<span>            xmax <span class="op">=</span> <span class="fl">436.089</span>, ymin <span class="op">=</span> <span class="fl">124.009</span>, </span>
<span>            ymax <span class="op">=</span> <span class="fl">359.009</span><span class="op">)</span>, coords_img <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="fl">362.716</span>, </span>
<span>            xmax <span class="op">=</span> <span class="fl">566.922</span>, ymin <span class="op">=</span> <span class="fl">161.212</span>, </span>
<span>            ymax <span class="op">=</span> <span class="fl">466.712</span><span class="op">)</span>, img_css_ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1.300</span>, </span>
<span>            y <span class="op">=</span> <span class="fl">1.299</span><span class="op">)</span>, mapping <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"X"</span>, y <span class="op">=</span> <span class="st">"Y"</span><span class="op">)</span>, </span>
<span>        domain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>left <span class="op">=</span> <span class="op">-</span><span class="fl">49.101</span>, right <span class="op">=</span> <span class="fl">57.228</span>, </span>
<span>            bottom <span class="op">=</span> <span class="op">-</span><span class="fl">70.389</span>, top <span class="op">=</span> <span class="fl">53.519</span><span class="op">)</span>, </span>
<span>        range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>left <span class="op">=</span> <span class="fl">50.986</span>, right <span class="op">=</span> <span class="fl">566.922</span>, </span>
<span>            bottom <span class="op">=</span> <span class="fl">603.013</span>, top <span class="op">=</span> <span class="fl">33.155</span><span class="op">)</span>, </span>
<span>        log <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>, direction <span class="op">=</span> <span class="st">"xy"</span>, </span>
<span>        brushId <span class="op">=</span> <span class="st">"ReducedDimensionPlot1_Brush"</span>, </span>
<span>        outputId <span class="op">=</span> <span class="st">"ReducedDimensionPlot1"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">red.panel</span><span class="op">[[</span><span class="st">"PanelWidth"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">4L</span> <span class="co"># To fit onto one line.</span></span>
<span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">red.panel</span>, <span class="va">diff.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Fdiffstat.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="feature-set-table">Feature set table<a class="anchor" aria-label="anchor" href="#feature-set-table"></a>
</h2>
<p>The <code><a href="../reference/FeatureSetTable-class.html">FeatureSetTable()</a></code> class is a bit unusual in that its
rows do not correspond to any dimension of the
<code>SummarizedExperiment</code>. Rather, each row is a feature set
(e.g., from GO or KEGG) that, upon click, transmits a multiple row
selection to other panels. The multiple selection consists of all rows
in the chosen feature set, allowing users to identify the positions of
all genes in a pathway of interest on, say, a volcano plot. This is also
a rare example of a panel that only transmits and does not receive any
selections from other panels.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/global-FeatureSetCommands.html">setFeatureSetCommands</a></span><span class="op">(</span><span class="fu"><a href="../reference/createGeneSetCommands.html">createGeneSetCommands</a></span><span class="op">(</span>identifier<span class="op">=</span><span class="st">"ENSEMBL"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gset.tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FeatureSetTable-class.html">FeatureSetTable</a></span><span class="op">(</span>Selected<span class="op">=</span><span class="st">"GO:0002576"</span>, </span>
<span>    Search<span class="op">=</span><span class="st">"platelet"</span>, PanelWidth<span class="op">=</span><span class="fl">6L</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># This volcano plot will highlight the genes in the selected gene set.</span></span>
<span><span class="va">vol.panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>RowSelectionSource<span class="op">=</span><span class="st">"FeatureSetTable1"</span>,</span>
<span>    ColorBy<span class="op">=</span><span class="st">"Row selection"</span>, PanelWidth<span class="op">=</span><span class="fl">6L</span><span class="op">)</span></span>
<span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gset.tab</span>, <span class="va">vol.panel</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="screenshots%2Fgeneset.png"><!-- --></p>
</div>
<div class="section level2">
<h2 id="app-modes">App modes<a class="anchor" aria-label="anchor" href="#app-modes"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em>
contains a number of “modes” that allow users to conveniently load an
<code>iSEE</code> instance in one of several common configurations:</p>
<ul>
<li>
<code><a href="../reference/modeEmpty.html">modeEmpty()</a></code> will launch an empty app, i.e., with no
panels. This is occasionally useful to jump to the landing page where a
user can then upload a <code>SummarizedExperiment</code> object.</li>
<li>
<code><a href="../reference/modeGating.html">modeGating()</a></code> will launch an app with multiple feature
assay panels that are linked to each other. This is useful for applying
sequential restrictions on the data, equivalent to gating in a flow
cytometry experiment.</li>
<li>
<code><a href="../reference/modeReducedDim.html">modeReducedDim()</a></code> will launch an app with multiple
reduced dimension plots. This is useful for examining different views of
large high-dimensional datasets (e.g., single-cell studies).</li>
</ul>
</div>
<div class="section level2">
<h2 id="miscellaneous-panels">Miscellaneous panels<a class="anchor" aria-label="anchor" href="#miscellaneous-panels"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em> also
includes a number of other panel types, that one can find useful within
different contexts.</p>
<p>Coupled to each chunk of code listed below, it is possible to display
a screenshot of the app showcasing them.</p>
<p><strong>AggregatedDotPlot</strong></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span></span>
<span>    <span class="va">sce</span>,</span>
<span>    initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="../reference/AggregatedDotPlot.html">AggregatedDotPlot</a></span><span class="op">(</span></span>
<span>            ColumnDataLabel<span class="op">=</span><span class="st">"Primary.Type"</span>,</span>
<span>            CustomRowsText <span class="op">=</span> <span class="st">"Rorb\nSnap25\nFoxp2"</span>,</span>
<span>            PanelHeight <span class="op">=</span> <span class="fl">500L</span>, </span>
<span>            PanelWidth <span class="op">=</span> <span class="fl">8L</span></span>
<span>        <span class="op">)</span>  </span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To be later run as...</span></span>
<span><span class="co"># app</span></span>
<span><span class="co">## ... or</span></span>
<span><span class="co"># shiny::runApp(app)</span></span></code></pre></div>
<p><img src="screenshots%2Faggrodotplot.png"><!-- --></p>
<p>This can be very useful as an alternative to the
<code>ComplexHeatmapPlot</code> panel, as sometimes it is not just about
the shifts in average expression levels, but true biological signal can
be found e.g. in scenarios such as differential detection.</p>
<p><strong>MarkdownBoard</strong></p>
<p>The <code>MarkdownBoard</code> panel class renders Markdown notes,
user-supplied, into HTML to display inside the app.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span></span>
<span>    <span class="va">sce</span>, </span>
<span>    initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="../reference/MarkdownBoard-class.html">MarkdownBoard</a></span><span class="op">(</span></span>
<span>            Content <span class="op">=</span> <span class="st">"# `iSEE` notepad\n\nYou can enter anything here.\n\nA list of marker genes you might be interested into:\n\n- Snap25\n- Rorb\n- Foxp2\n\nThis makes it easier to copy-paste while staying inside `iSEE`.  \nAs you can notice, the full power of markdown is at your service.\n\nHave fun exploring your data, in an even more efficient manner!\n"</span>, </span>
<span>            PanelWidth <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>            DataBoxOpen <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To be later run as...</span></span>
<span><span class="co"># app</span></span>
<span><span class="co">## ... or</span></span>
<span><span class="co"># shiny::runApp(app)</span></span></code></pre></div>
<p>This is useful for displaying information alongside other panels, or
for users to simply jot down their own notes (and re-use them more
efficiently later).</p>
<p><img src="screenshots%2Fmarkdownboard.png"><!-- --></p>
<p>The content of the <code>MarkdownBoard</code> is included in the
<code>Data parameters</code> portion of the panel, as visible in the
screenshot.</p>
<p><em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> will
take care of rendering your notes into good-looking yet simple HTML,
that can embedded in a variety of analytic workflows for the data under
inspection.</p>
</div>
<div class="section level2">
<h2 id="contributing-to-iseeu">Contributing to <em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em><a class="anchor" aria-label="anchor" href="#contributing-to-iseeu"></a>
</h2>
<p>If you want to contribute to the development of the <em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em>
package, here is a quick step-by-step guide:</p>
<ul>
<li>Fork the <code>iSEEu</code> repository from GitHub (<a href="https://github.com/iSEE/iSEEu" class="external-link uri">https://github.com/iSEE/iSEEu</a>) and clone it
locally.</li>
</ul>
<pre><code>git clone https://github.com/[your_github_username]/iSEEu.git</code></pre>
<ul>
<li><p>Add the desired new files - start from the <code>R</code> folder,
then document via <code>roxygen2</code> - and push to your fork. As an
example you can check out to understand how things are supposed to work,
there are several modes already defined in the <code>R/</code>
directory. A typical contribution could include e.g. a function defining
an <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> mode,
named <code>modeXXX</code>, where <code>XXX</code> provides a clear
representation of the purpose of the mode. Please place each mode in a
file of its own, with the same name as the function. The function should
be documented (including an example), and any required packages should
be added to the <code>DESCRIPTION</code> file.</p></li>
<li><p>Once your <code>mode</code>/function is done, consider adding
some information in the package. Some examples might be a screenshot of
the mode in action (to be placed in the folder
<code>inst/modes_img</code>), and a well-documented example use case
(maybe an entry in the <code>vignettes</code> folder). Also add yourself
as a contributor (<code>ctb</code>) to the <code>DESCRIPTION</code>
file.</p></li>
<li><p>Make a pull request to the original repo - the GitHub site offers
a practical framework to do so, enabling comments, code reviews, and
other goodies. The <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> core
team will evaluate the contribution and get back to you!</p></li>
</ul>
<p>That’s pretty much it!</p>
<div class="section level5">
<h5 class="unnumbered" id="using-example-data-sets">Using example data sets<a class="anchor" aria-label="anchor" href="#using-example-data-sets"></a>
</h5>
<p>Example data sets can often be obtained from an <em><a href="https://bioconductor.org/packages/3.20/ExperimentHub" class="external-link">ExperimentHub</a></em>
package (e.g. from the <em><a href="https://bioconductor.org/packages/3.20/scRNAseq" class="external-link">scRNAseq</a></em>
package for single-cell RNA-sequencing data), and should not be added to
the <em><a href="https://bioconductor.org/packages/3.20/iSEEu" class="external-link">iSEEu</a></em>
package.</p>
</div>
<div class="section level5">
<h5 class="unnumbered" id="documenting-testing-coding-style-and-conventions">Documenting, testing, coding style and
conventions<a class="anchor" aria-label="anchor" href="#documenting-testing-coding-style-and-conventions"></a>
</h5>
<ul>
<li>If possible, please consider adding an example in the dedicated
Roxygen preamble to show how to run each function</li>
<li>If possible, consider adding one or more unit tests - we use the
<code>testthat</code> framework</li>
</ul>
<p>We do follow some guidelines regarding the names given to variables,
please abide to these for consistency with the rest of the codebase.
Here are a few pointers:</p>
<ul>
<li>Keep the indentation as it is in the initial functions already
available (4-spaces indentation).</li>
<li>If writing text (e.g. in the vignette), please use one sentence per
line - this makes <code>git diff</code> operations easier to check.</li>
<li>When choosing variable names, try to keep the consistency with what
already is existing:
<ul>
<li>
<code>camelCase</code> for modes and other functions</li>
<li>
<code>.function_name</code> for internals</li>
<li>
<code>PanelClassName</code> for panels</li>
<li>
<code>.genericFunction</code> for the API</li>
<li>
<code>.scope1.scope2.name</code> for variable names in the cached
info</li>
</ul>
</li>
</ul>
<p>If you intend to understand more in depth the internals of the <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>
framework, consider checking out the bookdown resource we put together
at <a href="https://isee.github.io/iSEE-book/" class="external-link uri">https://isee.github.io/iSEE-book/</a></p>
</div>
<div class="section level5">
<h5 class="unnumbered" id="looking-for-constants-within-isee">Looking for constants within <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em><a class="anchor" aria-label="anchor" href="#looking-for-constants-within-isee"></a>
</h5>
<p>Many of the “global” variables that are used in several places in
<em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>
are defined in the <a href="https://github.com/iSEE/iSEE/blob/master/R/constants.R" class="external-link">constants.R</a>
script in <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>. We
suggest to refer to those constants by their actual value rather than
their internal variable name in downstream panel code. Both constant
variable names and values may change at any time, but we will only
announce changes to the constant value.</p>
</div>
<div class="section level5">
<h5 class="unnumbered" id="what-if-i-need-a-custom-panel-type">What if I need a custom panel type?<a class="anchor" aria-label="anchor" href="#what-if-i-need-a-custom-panel-type"></a>
</h5>
<p>In addition to the eight standard panel types, custom panels are
easily accommodated within <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>
applications. For a guide, see the corresponding <a href="https://bioconductor.org/packages/release/bioc/vignettes/iSEE/inst/doc/custom.html" class="external-link">vignette</a>.
For examples, see <a href="https://github.com/iSEE/iSEE_custom" class="external-link">this
repo</a>.</p>
</div>
<div class="section level5">
<h5 class="unnumbered" id="where-can-i-find-a-comprehensive-introduction-to-isee">Where can I find a comprehensive introduction to
<em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>?<a class="anchor" aria-label="anchor" href="#where-can-i-find-a-comprehensive-introduction-to-isee"></a>
</h5>
<p>The <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> package
contains several vignettes detailing the main functionality. You can
also take a look at this <a href="https://iSEE.github.io/iSEEWorkshop2019/index.html" class="external-link">workshop</a>.
A compiled version from the Bioc2019 conference (based on Bioconductor
release 3.10) is available <a href="http://biocworkshops2019.bioconductor.org.s3-website-us-east-1.amazonaws.com/page/iSEEWorkshop2019__iSEE-lab/" class="external-link">here</a>.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] scater_1.33.4               ggplot2_3.5.1              </span></span>
<span><span class="co">##  [3] scuttle_1.15.4              scRNAseq_2.19.1            </span></span>
<span><span class="co">##  [5] edgeR_4.3.17                limma_3.61.12              </span></span>
<span><span class="co">##  [7] airway_1.25.0               iSEEu_1.17.1               </span></span>
<span><span class="co">##  [9] iSEEhex_1.7.1               iSEE_2.17.4                </span></span>
<span><span class="co">## [11] SingleCellExperiment_1.27.2 SummarizedExperiment_1.35.3</span></span>
<span><span class="co">## [13] Biobase_2.65.1              GenomicRanges_1.57.1       </span></span>
<span><span class="co">## [15] GenomeInfoDb_1.41.2         IRanges_2.39.2             </span></span>
<span><span class="co">## [17] S4Vectors_0.43.2            BiocGenerics_0.51.3        </span></span>
<span><span class="co">## [19] MatrixGenerics_1.17.0       matrixStats_1.4.1          </span></span>
<span><span class="co">## [21] BiocStyle_2.33.1           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.1            later_1.3.2              BiocIO_1.15.2           </span></span>
<span><span class="co">##   [4] bitops_1.0-9             filelock_1.0.3           tibble_3.2.1            </span></span>
<span><span class="co">##   [7] XML_3.99-0.17            lifecycle_1.0.4          httr2_1.0.5             </span></span>
<span><span class="co">##  [10] doParallel_1.0.17        lattice_0.22-6           ensembldb_2.29.1        </span></span>
<span><span class="co">##  [13] alabaster.base_1.5.9     magrittr_2.0.3           sass_0.4.9              </span></span>
<span><span class="co">##  [16] rmarkdown_2.28           jquerylib_0.1.4          yaml_2.3.10             </span></span>
<span><span class="co">##  [19] httpuv_1.6.15            DBI_1.2.3                RColorBrewer_1.1-3      </span></span>
<span><span class="co">##  [22] abind_1.4-8              zlibbioc_1.51.1          Rtsne_0.17              </span></span>
<span><span class="co">##  [25] AnnotationFilter_1.29.0  RCurl_1.98-1.16          rappdirs_0.3.3          </span></span>
<span><span class="co">##  [28] circlize_0.4.16          GenomeInfoDbData_1.2.13  ggrepel_0.9.6           </span></span>
<span><span class="co">##  [31] irlba_2.3.5.1            alabaster.sce_1.5.1      pkgdown_2.1.1.9000      </span></span>
<span><span class="co">##  [34] codetools_0.2-20         DelayedArray_0.31.14     DT_0.33                 </span></span>
<span><span class="co">##  [37] tidyselect_1.2.1         shape_1.4.6.1            UCSC.utils_1.1.0        </span></span>
<span><span class="co">##  [40] viridis_0.6.5            ScaledMatrix_1.13.0      shinyWidgets_0.8.7      </span></span>
<span><span class="co">##  [43] BiocFileCache_2.13.0     GenomicAlignments_1.41.0 jsonlite_1.8.9          </span></span>
<span><span class="co">##  [46] BiocNeighbors_1.99.2     GetoptLong_1.0.5         iterators_1.0.14        </span></span>
<span><span class="co">##  [49] systemfonts_1.1.0        foreach_1.5.2            tools_4.4.1             </span></span>
<span><span class="co">##  [52] ragg_1.3.3               Rcpp_1.0.13              glue_1.8.0              </span></span>
<span><span class="co">##  [55] gridExtra_2.3            SparseArray_1.5.44       xfun_0.48               </span></span>
<span><span class="co">##  [58] mgcv_1.9-1               dplyr_1.1.4              HDF5Array_1.33.8        </span></span>
<span><span class="co">##  [61] gypsum_1.1.6             shinydashboard_0.7.2     withr_3.0.1             </span></span>
<span><span class="co">##  [64] BiocManager_1.30.25      fastmap_1.2.0            rhdf5filters_1.17.0     </span></span>
<span><span class="co">##  [67] fansi_1.0.6              shinyjs_2.1.0            rsvd_1.0.5              </span></span>
<span><span class="co">##  [70] digest_0.6.37            R6_2.5.1                 mime_0.12               </span></span>
<span><span class="co">##  [73] textshaping_0.4.0        colorspace_2.1-1         listviewer_4.0.0        </span></span>
<span><span class="co">##  [76] RSQLite_2.3.7            utf8_1.2.4               generics_0.1.3          </span></span>
<span><span class="co">##  [79] hexbin_1.28.4            rtracklayer_1.65.0       httr_1.4.7              </span></span>
<span><span class="co">##  [82] htmlwidgets_1.6.4        S4Arrays_1.5.10          pkgconfig_2.0.3         </span></span>
<span><span class="co">##  [85] gtable_0.3.5             blob_1.2.4               ComplexHeatmap_2.21.1   </span></span>
<span><span class="co">##  [88] XVector_0.45.0           htmltools_0.5.8.1        bookdown_0.40           </span></span>
<span><span class="co">##  [91] ProtGenerics_1.37.1      rintrojs_0.3.4           clue_0.3-65             </span></span>
<span><span class="co">##  [94] scales_1.3.0             alabaster.matrix_1.5.10  png_0.1-8               </span></span>
<span><span class="co">##  [97] knitr_1.48               rjson_0.2.23             nlme_3.1-166            </span></span>
<span><span class="co">## [100] curl_5.2.3               shinyAce_0.4.2           cachem_1.1.0            </span></span>
<span><span class="co">## [103] rhdf5_2.49.0             GlobalOptions_0.1.2      BiocVersion_3.20.0      </span></span>
<span><span class="co">## [106] parallel_4.4.1           miniUI_0.1.1.1           vipor_0.4.7             </span></span>
<span><span class="co">## [109] AnnotationDbi_1.67.0     restfulr_0.0.15          desc_1.4.3              </span></span>
<span><span class="co">## [112] pillar_1.9.0             grid_4.4.1               alabaster.schemas_1.5.0 </span></span>
<span><span class="co">## [115] vctrs_0.6.5              promises_1.3.0           BiocSingular_1.21.4     </span></span>
<span><span class="co">## [118] dbplyr_2.5.0             beachmat_2.21.6          xtable_1.8-4            </span></span>
<span><span class="co">## [121] cluster_2.1.6            beeswarm_0.4.0           evaluate_1.0.0          </span></span>
<span><span class="co">## [124] GenomicFeatures_1.57.1   cli_3.6.3                locfit_1.5-9.10         </span></span>
<span><span class="co">## [127] compiler_4.4.1           Rsamtools_2.21.2         rlang_1.1.4             </span></span>
<span><span class="co">## [130] crayon_1.5.3             ggbeeswarm_0.7.2         fs_1.6.4                </span></span>
<span><span class="co">## [133] viridisLite_0.4.2        alabaster.se_1.5.3       BiocParallel_1.39.0     </span></span>
<span><span class="co">## [136] munsell_0.5.1            Biostrings_2.73.2        lazyeval_0.2.2          </span></span>
<span><span class="co">## [139] colourpicker_1.3.0       Matrix_1.7-0             ExperimentHub_2.13.1    </span></span>
<span><span class="co">## [142] bit64_4.5.2              Rhdf5lib_1.27.0          KEGGREST_1.45.1         </span></span>
<span><span class="co">## [145] statmod_1.5.0            shiny_1.9.1              alabaster.ranges_1.5.2  </span></span>
<span><span class="co">## [148] highr_0.11               AnnotationHub_3.13.3     fontawesome_0.5.2       </span></span>
<span><span class="co">## [151] igraph_2.0.3             memoise_2.0.1            bslib_0.8.0             </span></span>
<span><span class="co">## [154] bit_4.5.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-iSEE-2018" class="csl-entry">
Rue-Albrecht, Kevin, Federico Marini, Charlotte Soneson, and Aaron T. L.
Lun. 2018. <span>“iSEE: Interactive SummarizedExperiment
Explorer.”</span> <em>F1000Research</em> 7 (June): 741. <a href="https://doi.org/10.12688/f1000research.14966.1" class="external-link">https://doi.org/10.12688/f1000research.14966.1</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kevin Rue-Albrecht, Charlotte Soneson, Federico Marini, Aaron Lun.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
