<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Universe of iSEE panels • iSEEu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Universe of iSEE panels">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/iSEEu_contribute_eurobioc2019.html">iSEEu_contribute_eurobioc2019
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/iSEE/iSEEu">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Universe of iSEE panels</h1>
                        <h4 class="author">Aaron Lun</h4>
                  <a class="author_email" href="mailto:#"></a><a href="mailto:infinite.monkeys.with.keyboards@gmail.com">infinite.monkeys.with.keyboards@gmail.com</a>
      
                  
            <h4 class="date">15 March 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEu/blob/master/vignettes/universe.Rmd"><code>vignettes/universe.Rmd</code></a></small>
      <div class="hidden name"><code>universe.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.11/iSEE">iSEE</a></em> package provides a general and flexible framework for interactively exploring <code>SummarizedExperiment</code> objects. However, in many cases, more specialized panels are required for effective visualization of specific data types. The <em><a href="https://bioconductor.org/packages/3.11/iSEEu">iSEEu</a></em> package implements a collection of such dedicated panel classes that work directly in the <code>iSEE</code> application and can smoothly interact with other panels. This allows users to quickly parametrize bespoke apps for their data to address scientific questions of interest. We first load in the package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(iSEEu)</a></code></pre></div>
<p>All the panels described in this document can be specified by simply passing them into the <code>iSEE()</code> function via the <code>initial=</code> argument. For example, we can create an app to explore differential expression results using a volcano and MA plot:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Making up some results:</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">se &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(<span class="kw"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(<span class="dv">10000</span>), <span class="dv">1000</span>, <span class="dv">10</span>))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(se) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"GENE_"</span>, <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(se)))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">rowData</span>(se)<span class="op">$</span>PValue &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(se))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">rowData</span>(se)<span class="op">$</span>LogFC &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(se))</a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="kw">rowData</span>(se)<span class="op">$</span>AveExpr &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(se))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">app &lt;-<span class="st"> </span><span class="kw">iSEE</span>(se, <span class="dt">initial=</span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span>(), <span class="kw"><a href="../reference/MAPlot-class.html">MAPlot</a></span>()))</a></code></pre></div>
</div>
<div id="differential-expression-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression-plots" class="anchor"></a>Differential expression plots</h1>
<p>The <code>MAPlot</code> class creates a MA plot, i.e., with the log-fold change on the y-axis and the average expression on the x-axis. Features with significant differences in each direction are highlighted and counted on the legend. Users can vary the significance threshold and apply <em>ad hoc</em> filters on the log-fold change. This is a subclass of the <code>RowDataPlot</code> so points can be transmitted to other panels as multiple row selections. Instances of this class are created like:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">ma.panel &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MAPlot-class.html">MAPlot</a></span>()</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(ma.panel)</a></code></pre></div>
<pre><code>## [1] "MAPlot"
## attr(,"package")
## [1] "iSEEu"</code></pre>
<p>The <code>VolcanoPlot</code> class creates a volcano plot with the log-fold change on the x-axis and the negative log-p-value on the y-axis. Features with significant differences in each direction are highlighted and counted on the legend. Users can vary the significance threshold and apply <em>ad hoc</em> filters on the log-fold change. This is a subclass of the <code>RowDataPlot</code> so points can be transmitted to other panels as multiple row selections. Instances of this class are created like:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">vol.panel &lt;-<span class="st"> </span><span class="kw"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/class.html">class</a></span>(vol.panel)</a></code></pre></div>
<pre><code>## [1] "VolcanoPlot"
## attr(,"package")
## [1] "iSEEu"</code></pre>
</div>
<div id="app-modes" class="section level1">
<h1 class="hasAnchor">
<a href="#app-modes" class="anchor"></a>App modes</h1>
<p><em><a href="https://bioconductor.org/packages/3.11/iSEEu">iSEEu</a></em> contains a number of “modes” that allow users to conveniently load an <code>iSEE</code> instance in one of several common configurations:</p>
<ul>
<li>
<code><a href="../reference/modeEmpty.html">modeEmpty()</a></code> will launch an empty app, i.e., with no panels. This is occasionally useful to jump to the landing page where a user can then upload a <code>SummarizedExperiment</code> object.</li>
<li>
<code><a href="../reference/modeGating.html">modeGating()</a></code> will launch an app with multiple feature assay panels that are linked to each other. This is useful for applying sequential restrictions on the data, equivalent to gating in a flow cytometry experiment.</li>
<li>
<code><a href="../reference/modeReducedDim.html">modeReducedDim()</a></code> will launch an app with multiple reduced dimension plots. This is useful for examining different views of large high-dimensional datasets (e.g., single-cell studies).</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R Under development (unstable) (2020-03-13 r77948)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.6 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] iSEEu_0.1.1                 iSEE_1.99.4                
##  [3] SingleCellExperiment_1.9.2  SummarizedExperiment_1.17.3
##  [5] DelayedArray_0.13.7         BiocParallel_1.21.2        
##  [7] matrixStats_0.56.0          Biobase_2.47.2             
##  [9] GenomicRanges_1.39.2        GenomeInfoDb_1.23.13       
## [11] IRanges_2.21.3              S4Vectors_0.25.13          
## [13] BiocGenerics_0.33.0         BiocStyle_2.15.6           
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-145           bitops_1.0-6           fs_1.3.1              
##  [4] RColorBrewer_1.1-2     rprojroot_1.3-2        tools_4.0.0           
##  [7] backports_1.1.5        R6_2.4.1               DT_0.12               
## [10] vipor_0.4.5            mgcv_1.8-31            colorspace_1.4-1      
## [13] GetoptLong_0.1.8       tidyselect_1.0.0       compiler_4.0.0        
## [16] shinyjs_1.1            desc_1.2.0             colourpicker_1.0      
## [19] bookdown_0.18          scales_1.1.0           pkgdown_1.4.1         
## [22] stringr_1.4.0          digest_0.6.25          rmarkdown_2.1         
## [25] XVector_0.27.1         pkgconfig_2.0.3        htmltools_0.4.0       
## [28] fastmap_1.0.1          htmlwidgets_1.5.1      rlang_0.4.5           
## [31] GlobalOptions_0.1.1    shiny_1.4.0.2          shape_1.4.4           
## [34] jsonlite_1.6.1         dplyr_0.8.5            RCurl_1.98-1.1        
## [37] magrittr_1.5           GenomeInfoDbData_1.2.2 Matrix_1.2-18         
## [40] Rcpp_1.0.3             munsell_0.5.0          lifecycle_0.2.0       
## [43] stringi_1.4.6          yaml_2.2.1             rintrojs_0.2.2        
## [46] MASS_7.3-51.5          zlibbioc_1.33.1        grid_4.0.0            
## [49] promises_1.1.0         shinydashboard_0.7.1   crayon_1.3.4          
## [52] miniUI_0.1.1.1         lattice_0.20-40        splines_4.0.0         
## [55] circlize_0.4.8         knitr_1.28             ComplexHeatmap_2.3.2  
## [58] pillar_1.4.3           igraph_1.2.4.2         rjson_0.2.20          
## [61] glue_1.3.2             evaluate_0.14          BiocManager_1.30.10   
## [64] png_0.1-7              httpuv_1.5.2           gtable_0.3.0          
## [67] purrr_0.3.3            clue_0.3-57            assertthat_0.2.1      
## [70] ggplot2_3.3.0          xfun_0.12              mime_0.9              
## [73] xtable_1.8-4           later_1.0.0            viridisLite_0.3.0     
## [76] tibble_2.1.3           memoise_1.1.0          cluster_2.1.0         
## [79] shinyWidgets_0.5.1     shinyAce_0.4.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#differential-expression-plots">Differential expression plots</a></li>
      <li><a href="#app-modes">App modes</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://csoneson.github.io">Charlotte Soneson</a>, <a href="https://federicomarini.github.io">Federico Marini</a>, <a href="https://kevinrue.github.io">Kevin Rue-Albrecht</a>, <a href="https://LTLA.github.io">Aaron Lun</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
